### DISCUSSION TITLE
Deleting from super large directories

### DISCUSSION THREAD

**Discussion by nivekuil** [seaweedfs/seaweedfs#1751] [Q&A] [answered]
Say I have an S3 bucket that is a super large directory. If I delete this S3 bucket, how does SeaweedFS know what to delete from the filer store? If it can find the filer entries to delete them, then could the list functionality still work in some other way?

**Comment by chrislusf** [comment, marked as answer]
Good question! SeaweedFS would not know which entries to delete. Even if it knows, deleting the entries one by one would take forever and not realistic.

Hopefully there should not be too many super large directories. Most likely you will have just one, if at all. 

If you need to create and later drop it, you can combine https://github.com/chrislusf/seaweedfs/wiki/Path-Specific-Filer-Store with the https://github.com/chrislusf/seaweedfs/wiki/Super-Large-Directories , so that the path-specific store can be dropped manually, and remove the collection manually.

**Reply by nivekuil** [reply]
That makes sense. It sounds like good practice to have a path specific store for each super large directory in general. Should be pretty easy, just a different cassandra keyspace for example.
